star/test_suite/starspots/src/run_star_extras.f90

```diff
@@ -0,0 +1,413 @@
+!
+!   mesa is free software; you can redistribute it and/or modify
+!   it under the terms of the gnu general library public license as published
+!   by the free software foundation; either version 2 of the license, or
+!   (at your option) any later version.
+!
+!   mesa is distributed in the hope that it will be useful, 
+!   but without any warranty; without even the implied warranty of
+!   merchantability or fitness for a particular purpose.  see the
+!   gnu library general public license for more details.
+!
+!   you should have received a copy of the gnu library general public license
+!   along with this software; if not, write to the free software
+!   foundation, inc., 59 temple place, suite 330, boston, ma 02111-1307 usa
+!
+! ***********************************************************************
+ 
+      module run_star_extras
+
+      use star_lib
+      use star_def
+      use const_def
+      use math_lib
+      use chem_def !! maybe taking up uncessary space but whatever
+      
+
+      implicit none
+      real(dp) :: fspot, xspot, PB_i, Teff_local, sb_sigma
+
+      include "test_suite_extras_def.inc"
+
+      contains
+
+      include "test_suite_extras.inc"
+   
+
+      ! these routines are called by the standard run_star check_model
+     ! contains
+      
+      subroutine extras_controls(id, ierr)
+         integer, intent(in) :: id
+         integer, intent(out) :: ierr
+         type (star_info), pointer :: s
+         ierr = 0
+         call star_ptr(id, s, ierr)
+         if (ierr /= 0) return
+         
+         ! this is the place to set any procedure pointers you want to change
+         ! e.g., other_wind, other_mixing, other_energy  (see star_data.inc)
+
+         ! the extras functions in this file will not be called
+         ! unless you set their function pointers as done below.
+         ! otherwise we use a null_ version which does nothing (except warn).
+
+
+         fspot = s% x_ctrl(1)
+         xspot = s% x_ctrl(2)
+
+
+         s% extras_startup => extras_startup
+         s% extras_start_step => extras_start_step
+         s% extras_check_model => extras_check_model
+         s% extras_finish_step => extras_finish_step
+         s% extras_after_evolve => extras_after_evolve
+         s% how_many_extra_history_columns => how_many_extra_history_columns
+         s% data_for_extra_history_columns => data_for_extra_history_columns
+         s% how_many_extra_profile_columns => how_many_extra_profile_columns
+         s% data_for_extra_profile_columns => data_for_extra_profile_columns  
+
+         s% how_many_extra_history_header_items => how_many_extra_history_header_items
+         s% data_for_extra_history_header_items => data_for_extra_history_header_items
+         s% how_many_extra_profile_header_items => how_many_extra_profile_header_items
+         s% data_for_extra_profile_header_items => data_for_extra_profile_header_items
+
+
+         if (s% use_other_surface_PT) then
```

Comment:
The `use_other_...` controls get handled by the main codebase, right? I think this is unnecessary. If it's been added because MESA _wasn't_ handling the logic, then that's worth investigating in the main code.

Reply:
Yes, good point. Should be safe to delete this extra logic. I'd say feel free to make that change.